src/one_dragon/base/screen/screen_loader.py

```
@@ -302,3 +356,54 @@ def update_current_screen_name(self, screen_name: str) -> None:
+    # reload() 后自动计算全局集合，Application 启动/停止时自动 enter/exit scope
+
+    def enter_scope(self, app_id: str) -> None:
+        """进入应用 scope，活跃范围 = 全局 screen + 该 app_id 的局部 screen
+
+        仅当 YAML 中存在 app_id 匹配的 screen 时才真正启用 scope，否则保持全量匹配。
+
+        Args:
+            app_id: 当前应用的唯一标识符（与 ScreenInfo.app_id 对应）
+        """
+        if not self._global_screen_names:
+            return  # 所有 screen 都没有 app_id，不启用 scope
```

Ruff (0.15.10)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)